### DISCUSSION TITLE
Could pulumi config set cause race condition ?

### DISCUSSION THREAD

**Discussion by yadsamit-disc** [pulumi/pulumi#7924] [Pulumi in Practice] [answered]
Issue Link: https://github.com/pulumi/pulumi/issues/7920#issue-990497472

Hi Team,

**Background:**
Our deployment pipeline is as follows: `git push by tag --> circleCi --> pulumi --> ECS fargate`

**Problem:**
Today we don't have a way to mark release/deployment version in our logs. In order to do so, I am trying to inject the git release version to circleCi and eventually to our monitoring platform agent (datadog here) through pulumi.

**Approach:**
I have found a way to do set the release version in the datadog agent. I am using the below command
`pulumi config set releaseVersion <<parameters.tag>>` in one of our circleCI config commands.

_Steps in the circleCi command:_

```
pulumi login to s3 bucket
pulumi config set releaseVersion <<parameters.tag>> in the relevant stack
pulumi up
pulumi logout
```

**Concerns:**
Would updating the stack config while deploying cause any issue - specially any race condition?

Thanks for your help.

**Comment by nimbinatus** [comment, marked as answer]
As noted in the [reply](https://github.com/pulumi/pulumi/issues/7920#issuecomment-915642005) on the attached issue, the outlined steps should be fine.
